R-package/R/xgb.DMatrix.R

```diff
@@ -1,13 +1,41 @@
 #' Construct xgb.DMatrix object
 #'
-#' Construct xgb.DMatrix object from either a dense matrix, a sparse matrix, or a local file.
-#' Supported input file formats are either a LIBSVM text file or a binary file that was created previously by
-#' \code{\link{xgb.DMatrix.save}}).
-#'
-#' @param data a \code{matrix} object (either numeric or integer), a \code{dgCMatrix} object,
-#'        a \code{dgRMatrix} object,
-#'        a \code{dsparseVector} object (only when making predictions from a fitted model, will be
-#'        interpreted as a row vector), or a character string representing a filename.
+#' Construct an 'xgb.DMatrix' object from a given data source, which can then be passed to functions
+#' such as \link{xgb.train} or \link{predict.xgb.Booster}.
+#'
+#' Function 'xgb.QuantileDMatrix' will construct a DMatrix with quantization for the histogram
+#' method already applied to it, which can be used to reduce memory usage (compared to using a
+#' a regular DMatrix first and then creating a quantization out of it) when using the histogram
+#' method (`tree_method = "hist"`, which is the default algorithm), but is not usable for the
+#' sorted-indices method (`tree_method = "exact"`).
+#' @param data Data from which to create a DMatrix, which can then be used for fitting models or
+#' for getting predictions out of a fitted model.
+#'
+#' Supported input types are as follows:\itemize{
+#' \item `matrix` objects, with types `numeric`, `integer`, or `logical`.
+#' \item `data.frame` objects, with columns of types `numeric`, `integer`, `logical`, or `factor`.
+#'
+#' If passing `enable_categorical=TRUE`, columns with `factor` type will be treated as categorical.
+#' Otherwise, if passing `enable_categorical=FALSE` and the data contains `factor` columns, an error
+#' will be thrown.
+#'
+#' Note that xgboost uses base-0 encoding for categorical types, hence `factor` types (which use base-1
+#' encoding') will be converted inside the function call. Be aware that the encoding used for `factor`
+#' types is not kept as part of the model, so in subsequent calls to `predict`, it is the user's
+#' responsibility to ensure that factor columns have the same levels as the ones from which the DMatrix
+#' was constructed.
+#'
+#' Other column types are not supported.
+#' \item CSR matrices, as class `dgRMatrix` from package `Matrix`.
+#' \item CSC matrices, as class `dgCMatrix` from package `Matrix`. These are \bold{not} supported for
+#' 'xgb.QuantileDMatrix'.
+#' \item Single-row CSR matrices, as class `dsparseVector` from package `Matrix`, which is interpreted
+#' as a single row (only when making predictions from a fitted model).
+#' \item Text files in SVMLight / LibSVM formats, passed as a path to the file. These are \bold{not}
```

Comment:
We can stop supporting it in DMatrix as well: https://github.com/dmlc/xgboost/issues/9472

Reply:
Are there definitive plans to remove it? I see it hasn't been removed from the C++ side yet.

Reply:
Not yet, I need to keep the work in sync with the work for federated learning, which uses text data for testing. But it's for certain that we want to deprecate it.

Reply:
Then let's better leave that for a subsequent PR after this and in-place predict have been merged, since both touch code sections dealing with this text format, like docs and such.